test/wallet-rpc-test.js

```diff
@@ -288,6 +289,55 @@ describe('Wallet RPC Methods', function() {
     });
   });
 
+  describe('gettransaction', () => {
+    it('should get a sent TX', async () => {
+      await wclient.execute('selectwallet', ['miner']);
+      const amount = 1.01020304;
+      const txid = await wclient.execute('sendtoaddress', [addressHot, amount]);
+
+      const wtx = await wclient.execute('gettransaction', [txid]);
+      assert.strictEqual(wtx.details.length, 1);
+      assert.strictEqual(wtx.details[0].category, 'send');
+      assert.strictEqual(wtx.details[0].address, addressHot);
+      assert.strictEqual(wtx.details[0].amount, amount * -1);
+      assert.strictEqual(wtx.amount, amount * -1);
+
+      // Check the fee is correct
+      const nodeTX = await nclient.execute('getrawtransaction', [txid, 1]);
+      let inValue = 0;
+      let outValue = 0;
+      for (const input of nodeTX.vin) {
+        const coin = await nclient.execute('getrawtransaction', [input.txid, 1]);
+        inValue += coin.vout[input.vout].value;
+      }
+      for (const output of nodeTX.vout) {
+        outValue += output.value;
+      }
+      // This will be negative, just like the wallet's TX details' fee
```

Comment:
Does bitcoin-core api call return this as negative number as well ?

Reply:
ah good catch - no, bitcoin-cli gettransaction returns a sent like this:
```
  "details": [
    {
      "account": "",
      "address": "<...>",
      "category": "send",
      "amount": -0.04416366,
      "vout": 0,
      "fee": 0.03659766,
      "abandoned": false
    },
    {
      "account": "",
      "address": "<...>",
      "category": "send",
      "amount": -0.00527500,
      "vout": 1,
      "fee": 0.03659766,
      "abandoned": false
    }
  ],
```

Reply:
actually i have another tx in my core wallet that looks different:

```
    {
      "account": "",
      "address": "<...>",
      "category": "send",
      "amount": -0.00163205,
      "label": "<...>",
      "vout": 0,
      "fee": -0.00000890,
      "abandoned": false
    },
```

Reply:
https://github.com/bitcoin/bitcoin/blob/4bfef0daeb9351c200b5bd09e60596a29b4d3523/src/wallet/rpcwallet.cpp#L1680-L1681

```
      "fee": x.xxx,                     (numeric) The amount of the fee in BTC. This is negative and only available for the 
                                           'send' category of transactions.
```

So... I don't know why that first TX returned a positive value from Core, maybe it was a tx from myself to myself or something.